source/m2mobject.cpp

```diff
@@ -482,7 +485,24 @@ sn_coap_hdr_s* M2MObject::handle_post_request(nsdl_s *nsdl,
                                         if(observation_handler) {
                                             observation_handler->value_updated(this);
                                         }
+                                        coap_response->options_list_ptr = (sn_coap_options_list_s*)malloc(sizeof(sn_coap_options_list_s));
+                                        memset(coap_response->options_list_ptr, 0, sizeof(sn_coap_options_list_s));
+
+                                        obj_name = M2MBase::name();
+                                        obj_name += String("/");
+                                        obj_inst_id = (char*)malloc(10);
+                                        snprintf(obj_inst_id, 10,"%d",instance_id);
+                                        obj_name += obj_inst_id;
+
+                                        coap_response->options_list_ptr->location_path_len = obj_name.length();
+                                        coap_response->options_list_ptr->location_path_ptr = (uint8_t*)malloc(coap_response->options_list_ptr->location_path_len);
```

Comment:
length should be checked for != 0 (undef behaviour in malloc)
Also if length > 0 newly created pointer must be NULL checked